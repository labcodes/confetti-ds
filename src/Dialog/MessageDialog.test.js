import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import { MessageDialog } from "./index";

const args = {
  title: "Sample Message Dialog",
  icon: "star",
  content:
    "This is the main content of a Message Dialog. You may want to put as much content as you want, as long as it's a string.",
  buttonProps: {
    text: "Required button",
    onClick: () => alert("Required button clicked"),
  },
  outlineButtonProps: {
    text: "Optional button",
    onClick: () => alert("Optional button clicked"),
  },
  isModal: false,
  isOpen: false,
  handleCLose: jest.fn(),
  isMessageDialog: true,
};

describe("MessageDialog", () => {
  it("renders with base props", () => {
    const Component = <MessageDialog {...args} />;
    const renderedComponent = renderer.create(Component).toJSON();

    expect(renderedComponent).toMatchSnapshot();
  });

  it("test with isOpen false", () => {
    const Component = (
      <MessageDialog {...args} isModal isLarge isOpen={false} />
    );
    const renderedComponent = renderer.create(Component).toJSON();

    expect(renderedComponent).toMatchSnapshot();
  });

  it("test with isModal false", () => {
    const Component = (
      <MessageDialog {...args} isModal={false} isLarge isOpen />
    );
    const renderedComponent = renderer.create(Component).toJSON();

    expect(renderedComponent).toMatchSnapshot();
  });

  it("test with isLarge false", () => {
    const Component = (
      <MessageDialog {...args} isModal isOpen isLarge={false} />
    );
    const renderedComponent = renderer.create(Component).toJSON();

    expect(renderedComponent).toMatchSnapshot();
  });

  it(`checks if isOpen false sets body overflow to `, () => {
    let isOpen = false;
    const handleClose = () => {
      isOpen = false;
    };

    const Component = (
      <MessageDialog
        {...args}
        isModal
        isOpen={isOpen}
        handleClose={handleClose}
      />
    );

    const renderedComponent = renderer.create(Component).toJSON();
    const wrapper = mount(Component);
    expect(document.body.style.overflow).toBe("hidden");

    expect(wrapper.find(".lab-dialog__title").exists()).toBeFalsy();
    expect(wrapper.find(".lab-dialog__body").exists()).toBeFalsy();

    expect(renderedComponent).toMatchSnapshot();
  });
  it("checks if isOpen true sets body overflow to hidden", () => {
    let isOpen = true;
    const handleClose = () => {
      isOpen = false;
    };
    const Component = (
      <MessageDialog
        {...args}
        isModal
        isOpen={isOpen}
        handleClose={handleClose}
      />
    );

    const renderedComponent = renderer.create(Component).toJSON();
    const wrapper = mount(Component);
    expect(document.body.style.overflow).toBe("hidden");
    expect(wrapper.find(".lab-dialog__title").text()).toBe(args.title);
    expect(wrapper.find(".lab-dialog__body").text()).toBe(args.content);

    expect(renderedComponent).toMatchSnapshot();
  });
  it("toggle isOpen", () => {
    const handleClose = () => {};
    const Component = (
      <MessageDialog {...args} isModal isOpen handleClose={handleClose} />
    );
    const renderedComponent = renderer.create(Component).toJSON();
    const wrapper = mount(Component);

    expect(document.body.style.overflow).toBe("hidden");
    expect(wrapper.find(".lab-dialog__title").text()).toBe(args.title);
    expect(wrapper.find(".lab-dialog__body").text()).toBe(args.content);

    wrapper.setProps({ isOpen: false });

    expect(document.body.style.overflow).toBe("unset");
    expect(wrapper.find(".lab-dialog__title").exists()).toBeFalsy();
    expect(wrapper.find(".lab-dialog__body").exists()).toBeFalsy();

    expect(renderedComponent).toMatchSnapshot();
  });
});
