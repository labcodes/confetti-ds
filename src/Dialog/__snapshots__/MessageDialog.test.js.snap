// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageDialog rendering with base props 1`] = `null`;

exports[`MessageDialog rendering with isLarge false 1`] = `
Array [
  <div
    className="lab-dialog-overlay"
    onClick={[Function]}
    role="presentation"
  />,
  <div
    className="lab-dialog lab-dialog--message
            "
  >
    <div
      aria-modal="true"
      className="lab-dialog__content lab-dialog__content--message"
      role="dialog"
    >
      <button
        className="lab-dialog__mobile-close-button lab-dialog__mobile-close-button--message"
        onClick={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        type="button"
      >
        <span
          className="lab-icon lab-icon--collapse-open"
        />
      </button>
      <div
        className="lab-dialog__header lab-dialog__header--message"
      >
        <button
          className="lab-dialog__close-button"
          onClick={[Function]}
          tabIndex="2"
          type="button"
        >
          <span
            className="lab-icon lab-icon--remove lab-dialog__close-button-icon"
          />
        </button>
      </div>
      <div
        className="lab-dialog__icon-wrapper"
      >
        <span
          className="lab-icon lab-icon--star lab-dialog__icon"
        />
      </div>
      <div
        className="lab-dialog__title lab-dialog__title--message"
      >
        Sample Message Dialog
      </div>
      <div
        className="lab-dialog__body lab-dialog__body--message"
      >
        This is the main content of a Message Dialog. You may want to put as much content as you want, as long as it's a string.
      </div>
      <div
        className="lab-dialog__footer lab-dialog__footer--message"
      >
        <button
          className="lab-btn lab-btn--outline lab-btn--normal"
          onClick={[Function]}
          tabIndex="3"
          type="button"
        >
          
          Optional button
        </button>
        <button
          className="lab-btn lab-btn--default lab-btn--normal"
          onClick={[Function]}
          tabIndex="1"
          type="button"
        >
          
          Required button
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`MessageDialog rendering with isModal false 1`] = `
<div
  className="lab-dialog lab-dialog--message
             lab-dialog--large"
>
  <div
    aria-modal="true"
    className="lab-dialog__content lab-dialog__content--message"
    role="dialog"
  >
    <button
      className="lab-dialog__mobile-close-button lab-dialog__mobile-close-button--message"
      onClick={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <span
        className="lab-icon lab-icon--collapse-open"
      />
    </button>
    <div
      className="lab-dialog__header lab-dialog__header--message"
    >
      <button
        className="lab-dialog__close-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className="lab-icon lab-icon--remove lab-dialog__close-button-icon"
        />
      </button>
    </div>
    <div
      className="lab-dialog__icon-wrapper"
    >
      <span
        className="lab-icon lab-icon--star lab-dialog__icon"
      />
    </div>
    <div
      className="lab-dialog__title lab-dialog__title--message"
    >
      Sample Message Dialog
    </div>
    <div
      className="lab-dialog__body lab-dialog__body--message"
    >
      This is the main content of a Message Dialog. You may want to put as much content as you want, as long as it's a string.
    </div>
    <div
      className="lab-dialog__footer lab-dialog__footer--message"
    >
      <button
        className="lab-btn lab-btn--outline lab-btn--normal"
        onClick={[Function]}
        type="button"
      >
        
        Optional button
      </button>
      <button
        className="lab-btn lab-btn--default lab-btn--normal"
        onClick={[Function]}
        type="button"
      >
        
        Required button
      </button>
    </div>
  </div>
</div>
`;

exports[`MessageDialog rendering with isOpen false 1`] = `null`;
